api/views/views.py

```diff
@@ -81,23 +83,15 @@ def get(self, request, format=None):
         return Response(out)
 
     def post(self, request, format=None):
+        jsonschema.validate(request.data, site_settings_schema)
         if "allow_registration" in request.data.keys():
             site_config.ALLOW_REGISTRATION = request.data["allow_registration"]
         if "allow_upload" in request.data.keys():
             site_config.ALLOW_UPLOAD = request.data["allow_upload"]
         if "skip_patterns" in request.data.keys():
             site_config.SKIP_PATTERNS = request.data["skip_patterns"]
         if "heavyweight_process" in request.data.keys():
-            HEAVYWEIGHT_PROCESS_ENV = request.data["heavyweight_process"]
-            HEAVYWEIGHT_PROCESS = (
-                int(HEAVYWEIGHT_PROCESS_ENV)
-                if HEAVYWEIGHT_PROCESS_ENV.isnumeric()
-                else 1
-            )
-            site_config.HEAVYWEIGHT_PROCESS = HEAVYWEIGHT_PROCESS
-        if "map_api_key" in request.data.keys():
-            site_config.MAP_API_KEY = request.data["map_api_key"]
```

Comment:
I think you lost map_api_key, when implementing the validating logic

Reply:
uh, good catch! I'll fix it this evening.

Reply:
one more reason to start thinking of automated tests